### ISSUE TITLE
.match('string1').match('string2') should do 'MATCH string1 MATCH string2' not 'MATCH string1, string2'

### ISSUE THREAD

**Issue by swist** [neo4jrb/neo4j#965]
As of neo4j 2.2 these two are not equivalent when matching something like so:

```
MATCH (c:Person)-[:KNOWS*0..{since: date}]->(:Person)-[:KNOWS{since:date2}]->(f:Person)
MATCH (c)-[r:KNOWS*1..]->(f)
```

A workaround is to do 
```
.match('(c:Person)-[:KNOWS*0..{since: date}]->(:Person)-[:KNOWS{since:date2}]->(f:Person)')
.with('c, f')
.match('(c:Person)-[r:KNOWS*1..]->(f)')
```

but it would be nice to have the API mirror the Neo4j syntax

**Comment by cheerfulstoic** [comment]
Yeah, agreed.  The original idea was to mirror ActiveRecord/arel, but Cypher is more dynamic that SQL.

Perhaps it isn't documented very well, but instead of using the `with` you can call `break` which will make the generated Cypher start fresh at that point like so:

```ruby
.match('(c:Person)-[:KNOWS*0..{since: date}]->(:Person)-[:KNOWS{since:date2}]->(f:Person)')
.break
.match('(c:Person)-[r:KNOWS*1..]->(f)')
```

Similarly you can use `with('*')` with I just thought about recently.  That can be useful when you actually do want a `WITH` and want to have a `MATCH` follow an `OPTIONAL MATCH` for example.

So question: Sometimes you want to combine `MATCH` clauses together and sometimes you don't.  Do you think `break` is enough to allow for this distinction (though keep in mind it breaks the whole query, not just the match).  Maybe being able to pass multiple arguments to `match` in enough to specify when you want them to be the same, but then are there cases where you would want to build a `Query` chain and pass it between functions and have the `MATCH` clauses combine?

Also, it might help if you could explain your understanding of the differences between combining and not combining `MATCH` clauses because I'm fuzzy on it ;)

**Comment by swist** [comment]
I have never noticed the neo4jrb implementaion until the query I was building in cypher and the query built by neo4jrb were not returning the same results.

I don't speak SQL so I am not the best person to explain it, but I remember reading a blog post (or a release note) about how neo4j have changed their execution planner to allow multiple start nodes. The big problem with neo4j 1.9 was that you had to figure out a sensible starting point, but now since we have multiple labels it's easier to do conjunctions to find sets of starting nodes (just find nodes in multiple labels)

Another advantage of having `MATCH` followed by `MATCH` is that due to this improvement (that I can't seem to find a link for, I'll dig some more later) the queries are a bit faster, today I saw a significant performance hit when I introduced 'WITH'

**Comment by cheerfulstoic** [comment]
Oh, I see, so basically there would always be an automatic `break` after the `query_as(:c)`?

**Comment by swist** [comment]
Yep, that would work! I haven't ran it against the test suite, though
